WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift

```diff
@@ -565,7 +565,10 @@ private extension DashboardViewController {
         }
 
         let hostingController = StoreOnboardingViewHostingController(viewModel: .init(isExpanded: false,
-                                                                                      siteID: site.siteID),
+                                                                                      siteID: site.siteID,
+                                                                                      whenNoTasksAvailable: { [weak self] in
+            self?.removeOnboardingCard()
```

Comment:
💭 this solution works, though I was wondering if we can show/hide the onboarding card using the one source of truth `DashboardViewModel.showOnboarding`. Maybe `StoreOnboardingViewModel` can be moved to `DashboardViewModel` similar to the feature announcement view model. Then, `DashboardViewModel.showOnboarding` can be determined by a combination of the UserDefaults and the state of the `StoreOnboardingViewModel` to hide it when `taskViewModels` is empty and `isRedacted` is `false`, or UserDefaults completed is true. This might require some refactoring and is just for your consideration, just thought it's easier to maintain/debug in the future if we have one source of truth for the visibility.

Reply:
Thank you for the idea, Jaclyn!
I agree that a single source of truth will make debugging easier. I have made the necessary changes. Could you take another look at the PR?